benchmarks/bench_ml.py

```diff
@@ -0,0 +1,451 @@
+
+    # calculate averages
+    for Algorithm in average_for_algo:
+        mean_obj_vals = np.mean(average_for_algo[Algorithm], axis=0)
+        minimums = np.min(mean_obj_vals, axis=1)
+        f_calls = np.argmin(mean_obj_vals, axis=1)
+
+        min_mean = np.mean(minimums)
+        min_stdd = np.std(minimums)
+        min_best = np.min(minimums)
+
+        f_mean = np.mean(f_calls)
+        f_stdd = np.std(f_calls)
+        f_best = np.min(f_calls)
+
+        def fmt(float_value):
+            return ("%.3f" % float_value)
+
+        output = "|".join([fmt(min_mean) + " +/- " + fmt(min_stdd)] + [fmt(v) for v in [ min_best, f_mean, f_stdd, f_best]])
+        result = table_template + output
+        print("")
+        print(Algorithm)
+        print(result)
+
+
+def evaluate_optimizer(base_estimator, model, dataset, max_iter, seed_value):
+    """
+    Evaluates some estimator for the task of optimization of parameters of some
+    model, given limited number of model evaluations.
+    Parentheses on parameters are used to be able to run function with pool.map method.
+
+    :param base_estimator: Estimator to use for optimization.
+    :param model: str, name of the ML model class to be used for parameter tuning
+    :param dataset: str, name of dataset to train ML model on
+    :param max_iter: a budget of evaluations
+    :param seed_value: random seed, used to set the random number generator in numpy
+    :return: a list of paris (p, f(p)), where p is a dictionary of the form "param name":value,
+            and f(p) is performance measure value achieved by the model for configuration p.
+            Such list contains history of execution of optimization.
+    """
+
+
+    # below seed is necessary for processes which fork at the same time
+    # so that random numbers generated in processes are different
+    np.random.seed(seed_value)
+
+    problem = TesterClass(model, dataset)
+    space = problem.model_class[MODEL_PARAMETERS]
+
+    dimensions = []
+
+    # convert space dictionary to dimensions
+    for k in space.keys():
+        v = space[k]
+
+        if v[VARIABLE_TYPE] == REAL_VARIABLE: # if real variable ...
+            dim_range = v[VARIABLE_RANGE]
+            dim = Real(dim_range[0], dim_range[1])
+        elif v[VARIABLE_TYPE] == INTEGER_VARIABLE: # if integer ...
+            dim_range = v[VARIABLE_RANGE]
+            dim = Integer(dim_range[0], dim_range[1])
+        elif v[VARIABLE_TYPE] == CATEGORY_VARIABLE: # if category ...
+            dim_range = v[VARIABLE_RANGE]
+            dim = Categorical(dim_range)
+        else:
+            raise BaseException("Unknown type of variable!")
+
+        dimensions.append((dim, k)) # need to remember names of dimensions for evaluation code
+
+    # initialization
+    estimator = base_estimator(random_state=seed_value)
+    solver = Optimizer([d[0] for d in dimensions], estimator, acq_optimizer="sampling", random_state=seed_value)
```

Comment:
The first one should be fixed now, you need to rebase to get the changes. For `forest_minimize` the default acquisition function is "sampling".